chopper/lib/src/interceptor.dart

```diff
@@ -52,8 +52,10 @@ class HeadersInterceptor implements RequestInterceptor {
       applyHeaders(request, headers);
 }
 
-typedef FutureOr<Response> ResponseInterceptorFunc<Value>(
+typedef ResponseInterceptorFunc = FutureOr<Response> Function<Value>(
     Response<Value> response);
+typedef DynamicResponseInterceptorFunc = FutureOr<Response> Function(
```

Comment:
didn't get why we need `DynamicResponseInterceptorFunc`, without passing a generic type `ResponseInterceptorFunc` it's already dynamic.

Reply:
Well the way you define the interceptor before was without a type argument, so the function definition will be strictly different from the new one.
So without the `DynamicResponseInterceptorFunc` you cannot specify an interceptor like: `FutureOr<Response> interceptor(Response r) {}` it has to be `FutureOr<Response<T>> interceptor<T>(Response<T> r) {}`

Reply:
`typedef ResponseInterceptorFunc<Value> = FutureOr<Response> Function(Response<Value> response);`
has a type argument which is optional so you should be able to do both

Reply:
No that definition of the interceptor does not allow you to use the body type in the actual interceptor

Reply:
hmm I see, `dartfmt --fix` generated somthing weird :/
`typedef ResponseInterceptorFunc<Value>` instead of `typedef ResponseInterceptorFunc = Function<Value>`